osbs/tekton.py

```diff
@@ -345,14 +346,19 @@ def log_and_sleep():
 
 
 class PipelineRun():
-    def __init__(self, os, pipeline_run_name, pipeline_run_data):
+    def __init__(self, os, pipeline_run_name, pipeline_run_data=None):
         self.os = os
         self.pipeline_run_name = pipeline_run_name
         self.api_path = 'apis'
         self.api_version = API_VERSION
-        self.data = pipeline_run_data
+        self.input_data = pipeline_run_data
         self._pipeline_run_url = None
 
+    @property
+    def data(self):
+        # always get fresh info
+        return self.get_info()
```

Comment:
Based on https://github.com/containerbuildsystem/osbs-client/pull/1034#discussion_r723143843, update_labels and update_annotations do not need latest info

Why does cancel need to check existence first? Wouldn't the HTTP call return 404 or something else that indicates the pipelineRun didn't exist? I'd like to avoid unnecessary HTTP calls when possible

Reply:
ok I'll re-check it